cockatrice/src/SpoilerBackgroundUpdaterThread.cpp

```diff
@@ -0,0 +1,249 @@
+    int lnTimeToWaitBetween = settingsCache->getDownloadSpoilerTimeMinutes() * 60000;
+
+    // How much time is needed until the next update (NOW() - Last Update Time == Time Remaining)
+    long lnLastUpdateTime = settingsCache->getDownloadSpoilerLastUpdateTime();
+    long lnTimeSinceLastUpdate = QDateTime::currentMSecsSinceEpoch() - lnLastUpdateTime;
+    if (lnTimeSinceLastUpdate <= 500 || lnLastUpdateTime == -1)
+    {
+        // The update just occurred less then half a sec ago, so we will restart the timer from the beginning (time zero)
+        mpTimerForSpoilers->start(lnTimeToWaitBetween);
+    }
+    else if (lnTimeSinceLastUpdate < lnTimeToWaitBetween)
+    {
+        // If we have checked for an update in the past and it's not yet time to check for another update
+        // Start the clock to count down until lnTimeSinceLastUpdate
+        mpTimerForSpoilers->start(static_cast<int>(lnTimeToWaitBetween - lnTimeSinceLastUpdate));
+    }
+    else
+    {
+        // It's been too long since we last checked (above threshold), so we will start now (time inf)
+        mpTimerForSpoilers->start(0);
+    }
+
+    qDebug() << "Spoiler Timer has started and will finish in" << mpTimerForSpoilers->interval()/60000. << "minutes";
+}
+
+void SpoilerBackgroundUpdaterThread::downloadSpoilersFile(QUrl acUrl)
+{
+    qDebug() << "Starting to download spoiler on NAM";
+    if (mpNetworkAccessManager == nullptr)
+    {
+        mpNetworkAccessManager = new QNetworkAccessManager(this);
+    }
+
+    QNetworkReply *lpNetworkReply = mpNetworkAccessManager->get(QNetworkRequest(acUrl));
+
+    connect(lpNetworkReply, SIGNAL(finished()), this, SLOT(actDownloadFinishedSpoilersFile()));
+}
+
+void SpoilerBackgroundUpdaterThread::downloadSpoilerSeasonStatusFile()
+{
+    qDebug() << "Checking spoiler season status";
+    if (mpNetworkAccessManager == nullptr)
+    {
+        mpNetworkAccessManager = new QNetworkAccessManager(this);
+    }
+
+    QNetworkReply *lpNetworkReply = mpNetworkAccessManager->get(QNetworkRequest(QString(GITHUB_SPOILERS_STATUS_URL)));
+
+    // Force the calling of actSeeIfSpoilerSeasonIsActive() in sequence instead of parallel
+    // as we need to get that result before we start this timer thread up.
+    QEventLoop lcSubLoop;
+    connect(lpNetworkReply, SIGNAL(finished()), this, SLOT(actSeeIfSpoilerSeasonIsActive()));
+    connect(lpNetworkReply, SIGNAL(finished()), &lcSubLoop, SLOT(quit()));
+    lcSubLoop.exec();
+}
+
+void SpoilerBackgroundUpdaterThread::actDownloadFinishedSpoilersFile()
+{
+    QNetworkReply *lpNetworkReply = static_cast<QNetworkReply *>(sender());
+    QNetworkReply::NetworkError leErrorCode = lpNetworkReply->error();
+    QString lsSpoilerSavePath = settingsCache->getSpoilerDatabasePath();
+    
+    if (leErrorCode != QNetworkReply::NoError)
+    {
+        QMessageBox::critical(nullptr, tr("Error"), tr("Network error: %1.").arg(lpNetworkReply->errorString()));
+    }
```

Comment:
Just a personal preference, take it as a "soft" suggestion:
In methods like this where you have a lot of if/then/else branches, try to keep the logic flow as simple as possible. For example, here adding a 
```
lpNetworkReply->deleteLater();
return;
```
you can remove the `else` branch and de-indent everything of one step.
The same can be done at line 185.
So instead of having three levels of nested `if`s, you can get an easier to follow flow like this:
```
if(!condition1)
  return;
if(!condition2)
  return;
get_the_job_done();
```